run-build-functions.sh

```diff
@@ -101,34 +101,47 @@ install_deps() {
 }
 
 restore_node_modules() {
-  local workspace_output
-  local workspace_exit_code
   local installer=$1
-  # YARN_IGNORE_PATH will ignore the presence of a local yarn executable (i.e. yarn 2) and default
-  # to using the global one (which, for now, is always yarn 1.x). See https://yarnpkg.com/configuration/yarnrc#ignorePath
-  # we can actually use this command for npm workspaces as well
-  workspace_output="$(YARN_IGNORE_PATH=1 yarn workspaces --json info 2>/dev/null)"
-  workspace_exit_code=$?
-  if [ $workspace_exit_code -eq 0 ]
-  then
-    echo "$installer workspaces detected"
-    local package_locations
-    # Extract all the packages and respective locations. .data will be a JSON object like
-    # {
-    #   "my-package-1": {
-    #     "location": "packages/blog-1",
-    #     "workspaceDependencies": [],
-    #     "mismatchedWorkspaceDependencies": []
-    #   },
-    #   (...)
-    # }
-    # We need to cache all the node_module dirs, or we'll always be installing them on each run
-    mapfile -t package_locations <<< "$(echo "$workspace_output" | jq -r '.data | fromjson | to_entries | .[].value.location')"
-    restore_js_workspaces_cache "${package_locations[@]}"
-  else
-    echo "No $installer workspaces detected"
-    restore_cwd_cache node_modules "node modules"
-  fi
+
+	if has_feature_flag "$featureFlags" "build-image_use_new_package_manager_detection"; then
```

Comment:
Please use the format below for consistency with the rest of the code base
```sh
if ...
then
  ...
fi
```

Reply:
putting the then after the if is more used inside the codebase I would argue that this is an inconsistency. (I prefer it to inline it as it conforms from the style with other programming languages to have not two lines + less lines = better readability)

But I would not bike-shedding about